network/src/connectivity_manager/builder.rs

```diff
@@ -29,8 +28,9 @@ pub type ConnectivityManagerService = ConnectivityManager<Fuse<Interval>, Expone
 /// The configuration fields for ConnectivityManager
 struct ConnectivityManagerBuilderConfig {
     network_context: Arc<NetworkContext>,
-    eligible: Arc<RwLock<HashMap<PeerId, x25519::PublicKey>>>,
-    seed_peers: HashMap<PeerId, Vec<NetworkAddress>>,
+    eligible: Arc<RwLock<HashMap<PeerId, HashSet<x25519::PublicKey>>>>,
+    seed_addrs: HashMap<PeerId, Vec<NetworkAddress>>,
+    seed_pubkey_sets: HashMap<PeerId, HashSet<x25519::PublicKey>>,
```

Comment:
what's the incentive to maintain both sets? We can pretty cheaply get the key from the address, no?

Reply:
for potential client/server model in the future, i want to avoid conflating pubkeys from addresses we can dial and pubkeys we'll accept for inbound connections. for example, maybe a server will only accept connections from some set of pubkeys, but doesn't have any addresses that it can dial (since it's a server).

Reply:
probably can use better naming to represent two sets like allowed inbound/outbound? but it also looks over optimization for now